doc/user_guide/docker/jupyter-hub.md

```diff
@@ -0,0 +1,188 @@
+
+To mitigate this scenario, [JupyterHub](https://jupyter.org/hub) was created. At a high level,
+it provides a web server that offers the following functionality:
+
+* Authentication of users against preconfigured methods (PAM, OAuth, Kerberos, etc.)
+* Allocation required resources (for example, starting virtual machines)
+* Spawning Jupyter processes for users.
+* Managing traffic between users and Jupyter processes.
+
+Together they provide a convenient and flexible open-source tool, which allows administrators to build 
+portals for corporate Jupyter users (for example, Data Science teams) or online courses (MOOCs).
+
+A more detailed description of the JupyterHub architecture can be found in 
+[A Conceptual Overview of JupyterHub](https://jupyterhub.readthedocs.io/en/stable/explanation/concepts.html) 
+document.
+
+## Custom AI Lab image
+
+As the first step in AI Lab integration with JupyterHub, we need to create a custom AI Lab Docker Image, 
+which has JupyterHub Python package installed in the AI Lab python environment.
+This step is required, as JupyterHub needs to plug into the Jupyter environment and communicate with  
+the running Jupyter processes inside the container.
+
+To create a custom AI Lab image, create a file named `Dockerfile` with the following contents:
+
+```dockerfile
+FROM exasol/ai-lab:latest
+USER root
+RUN /home/jupyter/jupyterenv/bin/python3 -m pip install jupyterhub==5.2.1
+USER ubuntu
+ENTRYPOINT []
+```
+
+Then run the command `docker build -t exasol/ai-lab-hub:latest .` (note the dot at the end of the command).
+
+After a couple of minutes, you should have the tagged image in your Docker:
+
+```shell
+$ docker images
+REPOSITORY              TAG       IMAGE ID       CREATED              SIZE
+exasol/ai-lab-hub       latest    8d20c95bc72a   5 seconds ago        3.94GB
+```
+
+## JupyterHub configuration
+
+Now it is time to instruct JupyterHub on how to launch our AI Lab instance.
+There are many options for how the Jupyter process can be started, but here we'll focus 
+on a relatively simple scenario: starting the Docker container on the same machine
+where JupyterHub is running.
+
+In JupyterHub's terminology, to allocate resources and start Jupyter, a special entity 
+called a "Spawner" is used. JupyterHub comes with many types of spawners, but in our 
+case `DockerSpawner` is the most suitable. As its name suggests, it uses docker to 
+run a specific instance each time a new user needs a Jupyter environment.
+
+To set up the `DockerSpawner`, you need to add the following lines to the central JupyterHub 
+configuration file `jupyterhub_config.py`:
+
+```python
+c.JupyterHub.spawner_class = "dockerspawner.DockerSpawner"
+c.JupyterHub.hub_connect_ip = 'jupyterhub'
+c.DockerSpawner.image = 'exasol/ai-lab-hub:latest'
+c.DockerSpawner.network_name = 'jupyterhub'
+c.DockerSpawner.remove = True
+c.DockerSpawner.volumes = {"/var/run/docker.sock": "/var/run/docker.sock"}
```

Comment:
We probably should create also a volume per user for the notebooks and data. And mention that the docker socket optional is, when user should be able to create docker container. Or, we don't show the docker socket at all, and say, using docker DB with Jupyterhub is not supported.